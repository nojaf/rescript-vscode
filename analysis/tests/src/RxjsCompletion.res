type keyPress =
  | Up(string)
  | Down(string)

@val
external window: {..} = "window"

let main = async () => {
  let keyMapObservable = {
    open Rxjs

    let keydown =
      fromEvent(Obj.magic(window), "keydown")->pipe2(
        map(event => Down(event["key"])),
        distinctUntilChanged(),
      )

    let keyup =
      fromEvent(Obj.magic(window), "keyup")->pipe2(
        map(event => Up(event["key"])),
        distinctUntilChanged(),
      )

    // merge(keydown, keyup).
    //                       ^com

	// Rxjs.Observable.subscribe, Rxjs.pipe and Rxjs.pipe2 should be completed
  }

  ignore(keyMapObservable)
}
