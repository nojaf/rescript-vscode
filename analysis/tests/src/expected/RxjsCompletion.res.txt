Complete src/RxjsCompletion.res 23:29
posCursor:[23:29] posNoWhite:[23:28] Found expr:[7:17->30:1]
posCursor:[23:29] posNoWhite:[23:28] Found expr:[8:2->29:26]
posCursor:[23:29] posNoWhite:[23:28] Found expr:[9:4->23:29]
posCursor:[23:29] posNoWhite:[23:28] Found expr:[11:4->23:29]
posCursor:[23:29] posNoWhite:[23:28] Found expr:[17:4->23:29]
posCursor:[23:29] posNoWhite:[23:28] Found expr:[23:7->23:29]
Pexp_field [23:7->23:28] _:[27:2->23:29]
Completable: Cpath Value[merge](Nolabel, Nolabel).""
Raw opens: 1 Rxjs.place holder
Package opens Pervasives.JsxModules.place holder
Resolved opens 2 pervasives Rxjs.res
ContextPath Value[merge](Nolabel, Nolabel).""
ContextPath Value[merge](Nolabel, Nolabel)
ContextPath Value[merge]
Path merge
CPPipe pathFromEnv:Observable found:true
Path Rxjs.Observable.
[{
    "label": "->Observable.subscribe",
    "kind": 12,
    "tags": [],
    "detail": "(t<'t>, 't => unit) => subscription",
    "documentation": null,
    "sortText": "subscribe",
    "textEdit": {
    "range": {"start": {"line": 23, "character": 28}, "end": {"line": 23, "character": 28}},
    "newText": "->Observable.subscribe"
  }
  }]

